slangpy/types/buffer.py

```diff
@@ -100,6 +104,62 @@ def resolve_element_type(program_layout: SlangProgramLayout, element_type: Any)
     return element_type
 
 
+def load_buffer_data_from_image(
+    path: Union[str, PathLike[str]],
+    flip_y: bool = False,
+    linearize: bool = False,
+    scale: float = 1.0,
+    offset: float = 0.0,
+    greyscale: bool = False,
+) -> np.ndarray[Any, Any]:
+    """
+    Helper to load an image from a file and convert it to a floating point tensor.
+    """ ""
+
+    # Load bitmap + convert to numpy array
+    bitmap = Bitmap(path)
+    data: np.ndarray[Any, Any] = np.asarray(bitmap)
+
+    # Validate array shape.
+    if data.ndim != 3:
+        raise ValueError(f"Bitmap data must be 3D, got {data.ndim}D")
+    if data.shape[2] not in [1, 2, 3, 4]:
+        raise ValueError(
+            f"Bitmap data must have 1, 2, 3 or 4 channels, got {data.shape[2]} channels"
+        )
+
+    # Flip if requested
+    if flip_y:
+        data = np.flipud(data)
+
+    # If not floating point, assume needs converting to normalized range.
+    if bitmap.component_type not in [
+        DataStruct.Type.float16,
+        DataStruct.Type.float32,
+        DataStruct.Type.float64,
+    ]:
+        data = data.astype(np.float32) / 255.0
+
+    # Optionally apply gammar correction.
+    if bitmap.srgb_gamma:
+        if linearize:
+            data = np.pow(data, 2.2)
+
+    # Optionally convert to single channel by averaging channels.
+    if greyscale:
```

Comment:
a lot of these conversion you can do with the `Bitmap` class
to go to grayscale you typically don't want just average but dot product with (0.2126, 0.7152, 0.0722) to get luminance